src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java

```diff
@@ -411,9 +472,16 @@ public void startElement(String uri,
                 throws SAXException {
             if (qName.equals(MODULE)) {
                 //create configuration
-                final String name = attributes.getValue(NAME);
+                final String originalName = attributes.getValue(NAME);
+                final String name;
+                if (threadModeSettings == null) {
```

Comment:
Talked about at https://github.com/checkstyle/checkstyle/pull/4420#discussion_r123916782 .
`null` was still left in. This shouldn't be `null` IMO unless you can point to a real specific reason.